core/services/gateway/handlers/functions/subscriptions.go

```diff
@@ -191,6 +205,14 @@ func (s *onchainSubscriptions) querySubscriptionsRange(ctx context.Context, bloc
 		subscriptionId := start + uint64(i)
 		subscription := subscription
 		s.subscriptions.UpdateSubscription(subscriptionId, &subscription)
+		err = s.orm.UpsertSubscription(CachedSubscription{
+			SubscriptionID:                      subscriptionId,
+			RouterContractAddress:               s.router.Address(),
+			IFunctionsSubscriptionsSubscription: subscription,
+		})
+		if err != nil {
+			s.lggr.Errorf("unexpected error updating subscription in the cache: %w", err)
+		}
```

Comment:
💭 We can persist this async if we think this might become a bottleneck

Reply:
Can we call Upsert only when the value changed? That would save us a lot of unnecessary DB calls.

Reply:
changed the subscription implementation to return a `true` when an actual update was performed, now it should only update the db when a value changes.